### PR TITLE
F* 2022.01.15

### PR DESCRIPTION
**Pull request by R1kM**
Release the latest version of F*
cc @tahina-pro

### CODE REVIEW

File: packages/fstar/fstar.2022.01.15/opam [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+authors: [
+  "Nik Swamy <nswamy@microsoft.com>"
+  "Jonathan Protzenko <protz@microsoft.com>"
+  "Tahina Ramananandro <taramana@microsoft.com>"
+]
+license: "Apache-2.0"
+homepage: "http://fstar-lang.org"
+bug-reports: "https://github.com/FStarLang/FStar/issues"
+depends: [
+  "ocaml" {>= "4.04.1" & < "4.13.0"}
```
Comment (kit-ty-kate) [inline packages/fstar/fstar.2022.01.15/opam:13]:
```suggestion
  "ocaml" {>= "4.04.1" & < "4.13.0"}
```
Is there any particular reason why this package is not compatible with 4.13?

Comment (kit-ty-kate) [reply packages/fstar/fstar.2022.01.15/opam:13]: Replying to my own question: there doesn’t seem to be any reason.
I’ve sent a patch upstream: https://github.com/FStarLang/FStar/pull/2434

Comment (R1kM) [reply packages/fstar/fstar.2022.01.15/opam]: Thanks!

**Comment by kit-ty-kate** [comment]
Fails on linux/arm64 and linux/s390x for some reason (worked fine locally on macOS/arm64). Any idea where that comes from?
```
#=== ERROR while compiling fstar.2022.01.15 ===================================#
# context              2.1.2 | linux/s390x | ocaml-base-compiler.4.13.1 | pinned(https://github.com/FStarLang/FStar/archive/refs/tags/v2022.01.15.tar.gz)
# path                 ~/.opam/4.13/.opam-switch/build/fstar.2022.01.15
# command              ~/.opam/opam-init/hooks/sandbox.sh build make PREFIX=/home/opam/.opam/4.13 -j 3
# exit-code            2
# env-file             ~/.opam/log/fstar-14404-c8092c.env
# output-file          ~/.opam/log/fstar-14404-c8092c.out
### output ###
# make[3]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make[2]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make[1]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib/ml'
# make[1]: Entering directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make FSTAR_HOME=.. -f Makefile.verify verify-core
# make[2]: Entering directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# ../bin/fstar.exe --warn_error @271-330   --use_hints --cache_dir .cache --hint_dir .cache --cache_checked_modules --odir _output --dep full  FStar.Algebra.CommMonoid.Equiv.fst FStar.Algebra.CommMonoid.fst FStar.Algebra.Monoid.fst FStar.All.fst FStar.BV.fst FStar.BigOps.fst FStar.BitVector.fst FStar.Calc.fst FStar.Classical.Sugar.fst FStar.Classical.fst FStar.DependentMap.fst FStar.Endianness.fst FStar.Exn.fst FStar.Fin.fst FStar.FunctionalExtensionality.fst FStar.GSet.fst FStar.Ghost.fst FStar.Heap.fst FStar.HyperStack.All.fst FStar.HyperStack.ST.fst FStar.HyperStack.fst FStar.IFC.fst FStar.IO.fst FStar.IndefiniteDescription.fst FStar.Int.Cast.Full.fst FStar.Int.Cast.fst FStar.Int.fst FStar.Int128.fst FStar.Int16.fst FStar.Int32.fst FStar.Int64.fst FStar.Int8.fst FStar.Integers.fst FStar.LexicographicOrdering.fst FStar.List.Pure.Base.fst FStar.List.Pure.Properties.fst FStar.List.Pure.fst FStar.List.Tot.Base.fst FStar.List.Tot.Properties.fst FStar.List.Tot.fst FStar.List.fst FStar.MRef.fst FStar.Map.fst FStar.MarkovsPrinciple.fst FStar.Math.Euclid.fst FStar.Math.Fermat.fst FStar.Math.Lemmas.fst FStar.Math.Lib.fst FStar.Modifies.fst FStar.ModifiesGen.fst FStar.Monotonic.DependentMap.fst FStar.Monotonic.Heap.fst FStar.Monotonic.HyperHeap.fst FStar.Monotonic.HyperStack.fst FStar.Monotonic.Map.fst FStar.Monotonic.Pure.fst FStar.Monotonic.Seq.fst FStar.Monotonic.Witnessed.fst FStar.Mul.fst FStar.Option.fst FStar.OrdSetProps.fst FStar.Order.fst FStar.PCM.fst FStar.PartialMap.fst FStar.Pervasives.Native.fst FStar.Pervasives.fst FStar.PredicateExtensionality.fst FStar.Preorder.fst FStar.Printf.fst FStar.PropositionalExtensionality.fst FStar.Ref.fst FStar.Reflection.Arith.fst FStar.Reflection.Const.fst FStar.Reflection.Derived.Lemmas.fst FStar.Reflection.Derived.fst FStar.Reflection.Formula.fst FStar.Reflection.fst FStar.ReflexiveTransitiveClosure.fst FStar.ST.fst FStar.Seq.Base.fst FStar.Seq.Permutation.fst FStar.Seq.Properties.fst FStar.Seq.Sorted.fst FStar.Seq.fst FStar.Set.fst FStar.Squash.fst FStar.SquashProperties.fst FStar.StrongExcludedMiddle.fst FStar.TSet.fst FStar.Tactics.Arith.fst FStar.Tactics.BV.fst FStar.Tactics.Canon.fst FStar.Tactics.CanonCommMonoid.fst FStar.Tactics.CanonCommMonoidSimple.Equiv.fst FStar.Tactics.CanonCommMonoidSimple.fst FStar.Tactics.CanonCommSemiring.fst FStar.Tactics.CanonCommSwaps.fst FStar.Tactics.CanonMonoid.fst FStar.Tactics.Derived.fst FStar.Tactics.Effect.fst FStar.Tactics.Logic.fst FStar.Tactics.PatternMatching.fst FStar.Tactics.Print.fst FStar.Tactics.Simplifier.fst FStar.Tactics.SyntaxHelpers.fst FStar.Tactics.Typeclasses.fst FStar.Tactics.Util.fst FStar.Tactics.fst FStar.UInt.fst FStar.UInt128.fst FStar.UInt16.fst FStar.UInt32.fst FStar.UInt64.fst FStar.UInt8.fst FStar.Universe.fst FStar.Util.fst FStar.Vector.Base.fst FStar.Vector.Properties.fst FStar.Vector.fst FStar.WellFounded.fst FStar.BV.fsti FStar.BaseTypes.fsti FStar.BigOps.fsti FStar.Bytes.fsti FStar.Char.fsti FStar.Classical.Sugar.fsti FStar.Classical.fsti FStar.Date.fsti FStar.DependentMap.fsti FStar.Dyn.fsti FStar.Endianness.fsti FStar.Float.fsti FStar.FunctionalExtensionality.fsti FStar.GSet.fsti FStar.Ghost.fsti FStar.HyperStack.ST.fsti FStar.IFC.fsti FStar.Int.fsti FStar.Int128.fsti FStar.Int16.fsti FStar.Int32.fsti FStar.Int64.fsti FStar.Int8.fsti FStar.LexicographicOrdering.fsti FStar.MRef.fsti FStar.Map.fsti FStar.Math.Euclid.fsti FStar.Math.Fermat.fsti FStar.Modifies.fsti FStar.ModifiesGen.fsti FStar.Monotonic.DependentMap.fsti FStar.Monotonic.Heap.fsti FStar.Monotonic.HyperHeap.fsti FStar.Monotonic.HyperStack.fsti FStar.Monotonic.Witnessed.fsti FStar.PartialMap.fsti FStar.Pervasives.fsti FStar.Range.fsti FStar.Real.fsti FStar.Reflection.Builtins.fsti FStar.Reflection.Data.fsti FStar.Reflection.Types.fsti FStar.ReflexiveTransitiveClosure.fsti FStar.Seq.Base.fsti FStar.Seq.Permutation.fsti FStar.Seq.Properties.fsti FStar.Set.fsti FStar.Squash.fsti FStar.String.fsti FStar.TSet.fsti FStar.Tactics.Builtins.fsti FStar.Tactics.Common.fsti FStar.Tactics.Effect.fsti FStar.Tactics.Result.fsti FStar.Tactics.Types.fsti FStar.Tcp.fsti FStar.UInt.fsti FStar.UInt128.fsti FStar.UInt16.fsti FStar.UInt32.fsti FStar.UInt64.fsti FStar.UInt8.fsti FStar.Udp.fsti FStar.Universe.fsti FStar.VConfig.fsti FStar.Vector.Base.fsti LowStar.Buffer.fst LowStar.BufferOps.fst LowStar.BufferView.Down.fst LowStar.BufferView.Up.fst LowStar.BufferView.fst LowStar.Comment.fst LowStar.ConstBuffer.fst LowStar.Endianness.fst LowStar.ImmutableBuffer.fst LowStar.Modifies.fst LowStar.ModifiesPat.fst LowStar.Monotonic.Buffer.fst LowStar.PrefixFreezableBuffer.fst LowStar.Printf.fst LowStar.RVector.fst LowStar.Regional.Instances.fst LowStar.Regional.fst LowStar.UninitializedBuffer.fst LowStar.Vector.fst LowStar.BufferView.Down.fsti LowStar.BufferView.Up.fsti LowStar.BufferView.fsti LowStar.Comment.fsti LowStar.ConstBuffer.fsti LowStar.Failure.fsti LowStar.Literal.fsti LowStar.Monotonic.Buffer.fsti LowStar.PrefixFreezableBuffer.fsti legacy/FStar.Array.fst legacy/FStar.Axiomatic.Array.fst legacy/FStar.Buffer.Quantifiers.fst legacy/FStar.Buffer.fst legacy/FStar.BufferNG.fst legacy/FStar.Constructive.fst legacy/FStar.Crypto.fst legacy/FStar.ErasedLogic.fst legacy/FStar.Error.fst legacy/FStar.HyperStack.IO.fst legacy/FStar.Pointer.Base.fst legacy/FStar.Pointer.Derived1.fst legacy/FStar.Pointer.Derived2.fst legacy/FStar.Pointer.Derived3.fst legacy/FStar.Pointer.fst legacy/FStar.Relational.Comp.fst legacy/FStar.Relational.Relational.fst legacy/FStar.TaggedUnion.fst legacy/FStar.TwoLevelHeap.fst legacy/LowStar.BufferCompat.fst legacy/LowStar.ToFStarBuffer.fst legacy/FStar.Array.fsti legacy/FStar.Matrix2.fsti legacy/FStar.Pointer.Base.fsti legacy/FStar.Pointer.Derived1.fsti legacy/FStar.Pointer.Derived2.fsti legacy/FStar.Pointer.Derived3.fsti legacy/FStar.TaggedUnion.fsti experimental/FStar.ConstantTime.Integers.fst experimental/FStar.InteractiveHelpers.Base.fst experimental/FStar.InteractiveHelpers.Effectful.fst experimental/FStar.InteractiveHelpers.ExploreTerm.fst experimental/FStar.InteractiveHelpers.Output.fst experimental/FStar.InteractiveHelpers.PostProcess.fst experimental/FStar.InteractiveHelpers.Propositions.fst experimental/FStar.InteractiveHelpers.fst experimental/FStar.MST.fst experimental/FStar.MSTTotal.fst experimental/FStar.NMST.fst experimental/FStar.NMSTTotal.fst experimental/FStar.OrdMap.fst experimental/FStar.OrdMapProps.fst experimental/FStar.OrdSet.fst experimental/FStar.Sequence.Ambient.fst experimental/FStar.Sequence.Base.fst experimental/FStar.Sequence.Permutation.fst experimental/FStar.Sequence.Seq.fst experimental/FStar.Sequence.Util.fst experimental/FStar.Sequence.fst experimental/FStar.Universe.PCM.fst experimental/Steel.Array.fst experimental/Steel.Channel.Protocol.fst experimental/Steel.Channel.Simplex.fst experimental/Steel.Closure.fst experimental/Steel.DisposableInvariant.fst experimental/Steel.Effect.Atomic.fst experimental/Steel.Effect.Common.fst experimental/Steel.Effect.M.fst experimental/Steel.Effect.fst experimental/Steel.FractionalAnchoredPreorder.fst experimental/Steel.FractionalPermission.fst experimental/Steel.GhostPCMReference.fst experimental/Steel.Heap.fst experimental/Steel.HigherReference.fst experimental/Steel.Loops.fst experimental/Steel.Memory.fst experimental/Steel.MonotonicCounter.fst experimental/Steel.MonotonicHigherReference.fst experimental/Steel.MonotonicReference.fst experimental/Steel.PCMMap.fst experimental/Steel.PCMReference.fst experimental/Steel.Preorder.fst experimental/Steel.Primitive.ForkJoin.Unix.fst experimental/Steel.Primitive.ForkJoin.fst experimental/Steel.Reference.fst experimental/Steel.ST.Array.fst experimental/Steel.ST.BitVector.fst experimental/Steel.ST.CancellableSpinLock.fst experimental/Steel.ST.Coercions.fst experimental/Steel.ST.Effect.Atomic.fst experimental/Steel.ST.Effect.AtomicAndGhost.fst experimental/Steel.ST.Effect.Ghost.fst experimental/Steel.ST.Effect.fst experimental/Steel.ST.EphemeralHashtbl.fst experimental/Steel.ST.GhostPCMReference.fst experimental/Steel.ST.GhostReference.fst experimental/Steel.ST.Loops.fst experimental/Steel.ST.MonotonicReference.fst experimental/Steel.ST.Reference.fst experimental/Steel.ST.SpinLock.fst experimental/Steel.ST.Util.fst experimental/Steel.Semantics.Hoare.MST.fst experimental/Steel.Semantics.Instantiate.fst experimental/Steel.SpinLock.fst experimental/Steel.Stepper.fst experimental/Steel.Utils.fst experimental/FStar.ConstantTime.Integers.fsti experimental/FStar.OrdMap.fsti experimental/FStar.OrdSet.fsti experimental/FStar.Sequence.Base.fsti experimental/FStar.Sequence.Permutation.fsti experimental/FStar.Sequence.Seq.fsti experimental/Steel.Array.fsti experimental/Steel.Channel.Duplex.fsti experimental/Steel.Channel.Simplex.fsti experimental/Steel.Closure.fsti experimental/Steel.DisposableInvariant.fsti experimental/Steel.Effect.Atomic.fsti experimental/Steel.Effect.Common.fsti experimental/Steel.Effect.fsti experimental/Steel.GhostPCMReference.fsti experimental/Steel.Heap.fsti experimental/Steel.HigherReference.fsti experimental/Steel.LockCoupling.fsti experimental/Steel.Loops.fsti experimental/Steel.Memory.fsti experimental/Steel.MonotonicHigherReference.fsti experimental/Steel.MonotonicReference.fsti experimental/Steel.PCMReference.fsti experimental/Steel.Primitive.ForkJoin.fsti experimental/Steel.Reference.fsti experimental/Steel.ST.Array.fsti experimental/Steel.ST.BitVector.fsti experimental/Steel.ST.CancellableSpinLock.fsti experimental/Steel.ST.Coercions.fsti experimental/Steel.ST.Effect.Atomic.fsti experimental/Steel.ST.Effect.AtomicAndGhost.fsti experimental/Steel.ST.Effect.Ghost.fsti experimental/Steel.ST.Effect.fsti experimental/Steel.ST.EphemeralHashtbl.fsti experimental/Steel.ST.GhostPCMReference.fsti experimental/Steel.ST.GhostReference.fsti experimental/Steel.ST.Loops.fsti experimental/Steel.ST.MonotonicReference.fsti experimental/Steel.ST.Reference.fsti experimental/Steel.ST.SpinLock.fsti experimental/Steel.ST.Util.fsti experimental/Steel.Semantics.Instantiate.fsti experimental/Steel.SpinLock.fsti > .depend
# [CHECK     prims.fst]
# Unexpected error; please file a bug report, ideally with a minimized version of the source program that triggered the error.
# (Failure "Empty option")
# make[2]: *** [Makefile.verify:52: .cache/prims.fst.checked] Error 1
# make[2]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make[1]: *** [Makefile:10: all] Error 2
# make[1]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make: *** [Makefile:8: all] Error 2
```

**Comment by kit-ty-kate** [comment]
Same for arm32 (with a slightly different error):
```
# make[3]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make[2]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make[1]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib/ml'
# make[1]: Entering directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make FSTAR_HOME=.. -f Makefile.verify verify-core
# make[2]: Entering directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# ../bin/fstar.exe --warn_error @271-330   --use_hints --cache_dir .cache --hint_dir .cache --cache_checked_modules --odir _output --dep full  FStar.Algebra.CommMonoid.Equiv.fst FStar.Algebra.CommMonoid.fst FStar.Algebra.Monoid.fst FStar.All.fst FStar.BV.fst FStar.BigOps.fst FStar.BitVector.fst FStar.Calc.fst FStar.Classical.Sugar.fst FStar.Classical.fst FStar.DependentMap.fst FStar.Endianness.fst FStar.Exn.fst FStar.Fin.fst FStar.FunctionalExtensionality.fst FStar.GSet.fst FStar.Ghost.fst FStar.Heap.fst FStar.HyperStack.All.fst FStar.HyperStack.ST.fst FStar.HyperStack.fst FStar.IFC.fst FStar.IO.fst FStar.IndefiniteDescription.fst FStar.Int.Cast.Full.fst FStar.Int.Cast.fst FStar.Int.fst FStar.Int128.fst FStar.Int16.fst FStar.Int32.fst FStar.Int64.fst FStar.Int8.fst FStar.Integers.fst FStar.LexicographicOrdering.fst FStar.List.Pure.Base.fst FStar.List.Pure.Properties.fst FStar.List.Pure.fst FStar.List.Tot.Base.fst FStar.List.Tot.Properties.fst FStar.List.Tot.fst FStar.List.fst FStar.MRef.fst FStar.Map.fst FStar.MarkovsPrinciple.fst FStar.Math.Euclid.fst FStar.Math.Fermat.fst FStar.Math.Lemmas.fst FStar.Math.Lib.fst FStar.Modifies.fst FStar.ModifiesGen.fst FStar.Monotonic.DependentMap.fst FStar.Monotonic.Heap.fst FStar.Monotonic.HyperHeap.fst FStar.Monotonic.HyperStack.fst FStar.Monotonic.Map.fst FStar.Monotonic.Pure.fst FStar.Monotonic.Seq.fst FStar.Monotonic.Witnessed.fst FStar.Mul.fst FStar.Option.fst FStar.OrdSetProps.fst FStar.Order.fst FStar.PCM.fst FStar.PartialMap.fst FStar.Pervasives.Native.fst FStar.Pervasives.fst FStar.PredicateExtensionality.fst FStar.Preorder.fst FStar.Printf.fst FStar.PropositionalExtensionality.fst FStar.Ref.fst FStar.Reflection.Arith.fst FStar.Reflection.Const.fst FStar.Reflection.Derived.Lemmas.fst FStar.Reflection.Derived.fst FStar.Reflection.Formula.fst FStar.Reflection.fst FStar.ReflexiveTransitiveClosure.fst FStar.ST.fst FStar.Seq.Base.fst FStar.Seq.Permutation.fst FStar.Seq.Properties.fst FStar.Seq.Sorted.fst FStar.Seq.fst FStar.Set.fst FStar.Squash.fst FStar.SquashProperties.fst FStar.StrongExcludedMiddle.fst FStar.TSet.fst FStar.Tactics.Arith.fst FStar.Tactics.BV.fst FStar.Tactics.Canon.fst FStar.Tactics.CanonCommMonoid.fst FStar.Tactics.CanonCommMonoidSimple.Equiv.fst FStar.Tactics.CanonCommMonoidSimple.fst FStar.Tactics.CanonCommSemiring.fst FStar.Tactics.CanonCommSwaps.fst FStar.Tactics.CanonMonoid.fst FStar.Tactics.Derived.fst FStar.Tactics.Effect.fst FStar.Tactics.Logic.fst FStar.Tactics.PatternMatching.fst FStar.Tactics.Print.fst FStar.Tactics.Simplifier.fst FStar.Tactics.SyntaxHelpers.fst FStar.Tactics.Typeclasses.fst FStar.Tactics.Util.fst FStar.Tactics.fst FStar.UInt.fst FStar.UInt128.fst FStar.UInt16.fst FStar.UInt32.fst FStar.UInt64.fst FStar.UInt8.fst FStar.Universe.fst FStar.Util.fst FStar.Vector.Base.fst FStar.Vector.Properties.fst FStar.Vector.fst FStar.WellFounded.fst FStar.BV.fsti FStar.BaseTypes.fsti FStar.BigOps.fsti FStar.Bytes.fsti FStar.Char.fsti FStar.Classical.Sugar.fsti FStar.Classical.fsti FStar.Date.fsti FStar.DependentMap.fsti FStar.Dyn.fsti FStar.Endianness.fsti FStar.Float.fsti FStar.FunctionalExtensionality.fsti FStar.GSet.fsti FStar.Ghost.fsti FStar.HyperStack.ST.fsti FStar.IFC.fsti FStar.Int.fsti FStar.Int128.fsti FStar.Int16.fsti FStar.Int32.fsti FStar.Int64.fsti FStar.Int8.fsti FStar.LexicographicOrdering.fsti FStar.MRef.fsti FStar.Map.fsti FStar.Math.Euclid.fsti FStar.Math.Fermat.fsti FStar.Modifies.fsti FStar.ModifiesGen.fsti FStar.Monotonic.DependentMap.fsti FStar.Monotonic.Heap.fsti FStar.Monotonic.HyperHeap.fsti FStar.Monotonic.HyperStack.fsti FStar.Monotonic.Witnessed.fsti FStar.PartialMap.fsti FStar.Pervasives.fsti FStar.Range.fsti FStar.Real.fsti FStar.Reflection.Builtins.fsti FStar.Reflection.Data.fsti FStar.Reflection.Types.fsti FStar.ReflexiveTransitiveClosure.fsti FStar.Seq.Base.fsti FStar.Seq.Permutation.fsti FStar.Seq.Properties.fsti FStar.Set.fsti FStar.Squash.fsti FStar.String.fsti FStar.TSet.fsti FStar.Tactics.Builtins.fsti FStar.Tactics.Common.fsti FStar.Tactics.Effect.fsti FStar.Tactics.Result.fsti FStar.Tactics.Types.fsti FStar.Tcp.fsti FStar.UInt.fsti FStar.UInt128.fsti FStar.UInt16.fsti FStar.UInt32.fsti FStar.UInt64.fsti FStar.UInt8.fsti FStar.Udp.fsti FStar.Universe.fsti FStar.VConfig.fsti FStar.Vector.Base.fsti LowStar.Buffer.fst LowStar.BufferOps.fst LowStar.BufferView.Down.fst LowStar.BufferView.Up.fst LowStar.BufferView.fst LowStar.Comment.fst LowStar.ConstBuffer.fst LowStar.Endianness.fst LowStar.ImmutableBuffer.fst LowStar.Modifies.fst LowStar.ModifiesPat.fst LowStar.Monotonic.Buffer.fst LowStar.PrefixFreezableBuffer.fst LowStar.Printf.fst LowStar.RVector.fst LowStar.Regional.Instances.fst LowStar.Regional.fst LowStar.UninitializedBuffer.fst LowStar.Vector.fst LowStar.BufferView.Down.fsti LowStar.BufferView.Up.fsti LowStar.BufferView.fsti LowStar.Comment.fsti LowStar.ConstBuffer.fsti LowStar.Failure.fsti LowStar.Literal.fsti LowStar.Monotonic.Buffer.fsti LowStar.PrefixFreezableBuffer.fsti legacy/FStar.Array.fst legacy/FStar.Axiomatic.Array.fst legacy/FStar.Buffer.Quantifiers.fst legacy/FStar.Buffer.fst legacy/FStar.BufferNG.fst legacy/FStar.Constructive.fst legacy/FStar.Crypto.fst legacy/FStar.ErasedLogic.fst legacy/FStar.Error.fst legacy/FStar.HyperStack.IO.fst legacy/FStar.Pointer.Base.fst legacy/FStar.Pointer.Derived1.fst legacy/FStar.Pointer.Derived2.fst legacy/FStar.Pointer.Derived3.fst legacy/FStar.Pointer.fst legacy/FStar.Relational.Comp.fst legacy/FStar.Relational.Relational.fst legacy/FStar.TaggedUnion.fst legacy/FStar.TwoLevelHeap.fst legacy/LowStar.BufferCompat.fst legacy/LowStar.ToFStarBuffer.fst legacy/FStar.Array.fsti legacy/FStar.Matrix2.fsti legacy/FStar.Pointer.Base.fsti legacy/FStar.Pointer.Derived1.fsti legacy/FStar.Pointer.Derived2.fsti legacy/FStar.Pointer.Derived3.fsti legacy/FStar.TaggedUnion.fsti experimental/FStar.ConstantTime.Integers.fst experimental/FStar.InteractiveHelpers.Base.fst experimental/FStar.InteractiveHelpers.Effectful.fst experimental/FStar.InteractiveHelpers.ExploreTerm.fst experimental/FStar.InteractiveHelpers.Output.fst experimental/FStar.InteractiveHelpers.PostProcess.fst experimental/FStar.InteractiveHelpers.Propositions.fst experimental/FStar.InteractiveHelpers.fst experimental/FStar.MST.fst experimental/FStar.MSTTotal.fst experimental/FStar.NMST.fst experimental/FStar.NMSTTotal.fst experimental/FStar.OrdMap.fst experimental/FStar.OrdMapProps.fst experimental/FStar.OrdSet.fst experimental/FStar.Sequence.Ambient.fst experimental/FStar.Sequence.Base.fst experimental/FStar.Sequence.Permutation.fst experimental/FStar.Sequence.Seq.fst experimental/FStar.Sequence.Util.fst experimental/FStar.Sequence.fst experimental/FStar.Universe.PCM.fst experimental/Steel.Array.fst experimental/Steel.Channel.Protocol.fst experimental/Steel.Channel.Simplex.fst experimental/Steel.Closure.fst experimental/Steel.DisposableInvariant.fst experimental/Steel.Effect.Atomic.fst experimental/Steel.Effect.Common.fst experimental/Steel.Effect.M.fst experimental/Steel.Effect.fst experimental/Steel.FractionalAnchoredPreorder.fst experimental/Steel.FractionalPermission.fst experimental/Steel.GhostPCMReference.fst experimental/Steel.Heap.fst experimental/Steel.HigherReference.fst experimental/Steel.Loops.fst experimental/Steel.Memory.fst experimental/Steel.MonotonicCounter.fst experimental/Steel.MonotonicHigherReference.fst experimental/Steel.MonotonicReference.fst experimental/Steel.PCMMap.fst experimental/Steel.PCMReference.fst experimental/Steel.Preorder.fst experimental/Steel.Primitive.ForkJoin.Unix.fst experimental/Steel.Primitive.ForkJoin.fst experimental/Steel.Reference.fst experimental/Steel.ST.Array.fst experimental/Steel.ST.BitVector.fst experimental/Steel.ST.CancellableSpinLock.fst experimental/Steel.ST.Coercions.fst experimental/Steel.ST.Effect.Atomic.fst experimental/Steel.ST.Effect.AtomicAndGhost.fst experimental/Steel.ST.Effect.Ghost.fst experimental/Steel.ST.Effect.fst experimental/Steel.ST.EphemeralHashtbl.fst experimental/Steel.ST.GhostPCMReference.fst experimental/Steel.ST.GhostReference.fst experimental/Steel.ST.Loops.fst experimental/Steel.ST.MonotonicReference.fst experimental/Steel.ST.Reference.fst experimental/Steel.ST.SpinLock.fst experimental/Steel.ST.Util.fst experimental/Steel.Semantics.Hoare.MST.fst experimental/Steel.Semantics.Instantiate.fst experimental/Steel.SpinLock.fst experimental/Steel.Stepper.fst experimental/Steel.Utils.fst experimental/FStar.ConstantTime.Integers.fsti experimental/FStar.OrdMap.fsti experimental/FStar.OrdSet.fsti experimental/FStar.Sequence.Base.fsti experimental/FStar.Sequence.Permutation.fsti experimental/FStar.Sequence.Seq.fsti experimental/Steel.Array.fsti experimental/Steel.Channel.Duplex.fsti experimental/Steel.Channel.Simplex.fsti experimental/Steel.Closure.fsti experimental/Steel.DisposableInvariant.fsti experimental/Steel.Effect.Atomic.fsti experimental/Steel.Effect.Common.fsti experimental/Steel.Effect.fsti experimental/Steel.GhostPCMReference.fsti experimental/Steel.Heap.fsti experimental/Steel.HigherReference.fsti experimental/Steel.LockCoupling.fsti experimental/Steel.Loops.fsti experimental/Steel.Memory.fsti experimental/Steel.MonotonicHigherReference.fsti experimental/Steel.MonotonicReference.fsti experimental/Steel.PCMReference.fsti experimental/Steel.Primitive.ForkJoin.fsti experimental/Steel.Reference.fsti experimental/Steel.ST.Array.fsti experimental/Steel.ST.BitVector.fsti experimental/Steel.ST.CancellableSpinLock.fsti experimental/Steel.ST.Coercions.fsti experimental/Steel.ST.Effect.Atomic.fsti experimental/Steel.ST.Effect.AtomicAndGhost.fsti experimental/Steel.ST.Effect.Ghost.fsti experimental/Steel.ST.Effect.fsti experimental/Steel.ST.EphemeralHashtbl.fsti experimental/Steel.ST.GhostPCMReference.fsti experimental/Steel.ST.GhostReference.fsti experimental/Steel.ST.Loops.fsti experimental/Steel.ST.MonotonicReference.fsti experimental/Steel.ST.Reference.fsti experimental/Steel.ST.SpinLock.fsti experimental/Steel.ST.Util.fsti experimental/Steel.Semantics.Instantiate.fsti experimental/Steel.SpinLock.fsti > .depend
# [CHECK     prims.fst]
# [CHECK     FStar.Pervasives.Native.fst]
# [CHECK     FStar.Pervasives.fsti]
# FStar.Pervasives.fsti(298,2-298,18): (Error 19) Subtyping check failed; expected type Prims.pure_wp a; got type p: Prims.pure_post a -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(303,2-303,27): (Error 19) Subtyping check failed; expected type Prims.pure_wp b; got type p: Prims.pure_post b -> Prims.Tot Prims.pure_pre; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(308,2-308,40): (Error 19) Subtyping check failed; expected type Prims.pure_wp a; got type post: Prims.pure_post a -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(313,2-313,19): (Error 19) Subtyping check failed; expected type Prims.pure_wp a; got type post: Prims.pure_post a -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(318,2-318,23): (Error 19) Subtyping check failed; expected type Prims.pure_wp a; got type p: Prims.pure_post a -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(323,2-323,17): (Error 19) Subtyping check failed; expected type Prims.pure_wp a; got type p: Prims.pure_post a -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(329,2-329,19): (Error 19) Subtyping check failed; expected type Prims.pure_wp Prims.unit; got type post: Prims.pure_post Prims.unit -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(335,2-335,19): (Error 19) Subtyping check failed; expected type Prims.pure_wp Prims.unit; got type post: Prims.pure_post Prims.unit -> Prims.Tot Prims.logical; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# FStar.Pervasives.fsti(367,25-367,58): (Error 19) assertion failed; The SMT solver could not prove the query, try to spell your proof in more detail or increase fuel/ifuel (see also prims.fst(339,38-339,60))
# make[2]: *** [Makefile.verify:52: .cache/FStar.Pervasives.fsti.checked] Error 1
# make[2]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make[1]: *** [Makefile:10: all] Error 2
# make[1]: Leaving directory '/home/opam/.opam/4.13/.opam-switch/build/fstar.2022.01.15/ulib'
# make: *** [Makefile:8: all] Error 2
```

**Comment by R1kM** [comment]
That's weird, off the top of my head, I can't see what would explain this different behaviour, and I also have no issue on macOS/arm64. @tahina-pro , does this ring a bell?
We tried relaxing the architecture constraint compared to the previous package especially to support macOS/arm, but we might need go back on this

**Comment by tahina-pro** [comment]
For the linux/arm32 case, I'd suspect Z3 behaving differently across platforms. For the linux/arm64 case, I have no clue.
Both cases were already reported at FStarLang/FStar#2332

**Comment by mseri** [comment]
This looks good. Is there anything blocking the merge or I can go on?

**Comment by R1kM** [comment]
It seems all CI failures are due to timeouts, it looks good to merge on my side.

**Pull request merged**